### ISSUE TITLE
404 Error for Hero page linked to existing Character article.

### ISSUE THREAD

**Issue by rachelvthomascoleman** [worldanvil/worldanvil-bug-tracker#29]
**World Anvil Username**: Malonromani

**World / Campaign / RPG Character / Manuscript Title**:  
World: "Malonromani's RPG Character Continuity"
Character with issue: "Sync fon Daath"
Character without issue, for reference: "Ion fon Daath"
Hero with issue: "Sync fon Daath"
Hero without issue, for reference: "Ion fon Daath"

**Feature**: 

**Related URLS** (To public & visible pages add screenshots otherwise): 
The character article for Sync: https://www.worldanvil.com/w/malonromani-s-rpg-character-continuity-malonromani/a/sync-fon-daath-person
The hero profile for Sync: https://www.worldanvil.com/hero/d7e21a51-ab9b-44d7-a05e-a9fe3130b3f7 
The character article for Ion: https://www.worldanvil.com/w/malonromani-s-rpg-character-continuity-malonromani/a/ion-fon-daath-person
The hero profile for Ion: https://www.worldanvil.com/hero/d7e21a51-ab9b-44d7-a05e-a9fe3130b3f7

**Describe the Issue**
A clear and concise description of what the issue is.

I was trying to find a way to include links between corresponding character articles and RPG hero profiles.  After clicking the button in the hero profile editor to connect it to my world, attempting to view the corresponding character article instead showed a 404 error page.  I thought I might have caused a conflict between my existing article and one that World Anvil might have tried to automatically create, so I copied the text of my character article to a text document and deleted both the hero profile and character article from World Anvil.  I then created a new hero profile, using the same character name, and clicked the button to connect it to my world.  This generated a new character article, which I then tried to view.  I still got the 404 error.

**To Reproduce**

1. I had created character articles in my World, "Malonromani's RPG Character Continuity", for characters named "Sync fon Daath" and "Ion fon Daath".
2. I had also created RPG hero profiles for characters with the same names.
3. I put World Anvil links in the format "[hero:1234-5678-...]" for the hero profiles into their matching character articles.
4. For some reason, the link to Sync's hero profile was not visible when I viewed the character article for Sync, but Ion's hero profile link was visible on Ion's character article.  I do not know if this contributed to the 404, but am including it just in case.
5. In the hero profile editor for Sync, under the "Character Access Rights" dropdown, I clicked a button that said "Connect to Malonromani's RPG Character Continuity".
6. I navigated back to the character article editor for Sync, and clicked the "View" button.
7. I was taken to a 404 error page.
8. I detached Sync's character sheet from his hero profile.
9. I deleted Sync's hero profile.
10. I deleted Sync's character article.
11. I created a new hero profile, using the same name "Sync fon Daath".
12. I attached Sync's character sheet to his hero profile.
13. In the hero profile editor for Sync, under the "Character Access Rights" dropdown, I clicked a button that said "Connect to Malonromani's RPG Character Continuity".
14. A character article for "Sync fon Daath" was created in my world.
15. I entered the character article editor for Sync, made sure that the article was set to "Published", and clicked the "View" button.
16. I was taken to the 404 error page.

**Expected behavior**

I expected the character article for Sync to be viewable, and to include a link to the hero profile.

**Screenshots**
If applicable, add screenshots to help explain your problem. You can add screenshots by drag & drop into this window!

**Comment by iamromeo** [comment]
@rachelvthomascoleman  Checking through the record the issue is that you have 2 articles in the same world with EXACTLY the same name and slug. 

This probably occurred because you had an article for that character and then one of your players was invited into the world with a hero with exactly the same name. 

Solution:
1. Go to your Articles & Categories
2. Search for the name of the article
3. Find the one NOT connected to the character profile
4. Delete the duplicate

**Comment by Kordishal** [comment]
Added a note to the Codex on how to prevent and resolve this issue.

**Comment by rachelvthomascoleman** [comment]
This worked, but the fact that the original article was still there might be a bug.

I was certain that I had deleted my original article for Sync, as I said in step 14 of my bug report.  I chose the "delete" option and the "confirm delete" option from the "Recently Edited Articles" list, and Sync's article stopped showing up in that list, and even the relationship between Sync and Ion was gone from Ion's article, so I thought Sync's article was gone.  However, when I clicked the link to show all character articles, it was still there alongside the one linked to the hero profile.  I again chose "delete" and "confirm delete" options, and this time the original article was deleted.
